### PR TITLE
Allow version arg to be optional in "kubeadm upgrade apply"

### LINKED ISSUE #460: kubeadm upgrade apply should read version from config file
version flag should be optional if --config is specified and kubernetesVersion key is specified in the file.

### PR DESCRIPTION
**Pull request by medinatiger**
**What this PR does / why we need it**:

This PR make the version arg optional if --config is specified and .KuberneteVersion is available.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
fixes https://github.com/kubernetes/kubeadm/issues/460

**Special notes for your reviewer**:
```release-note
Allow version arg in kubeadm upgrade apply to be optional if config file already have version info
```

### CODE REVIEW

**Review by jamiehannaford — APPROVED** [review]
LGTM. Defer to @luxas though

**Review by timothysc — APPROVED** [review]
I could nit but 
/lgtm 
/approve

**Comment by timothysc** [comment]
Added as cherry-pick candidate, b/c it fixes the UX for a deployed feature.

**Comment by timothysc** [comment]
@medinatiger please add a release note for this PR.

**Comment by kad** [comment]
it will be good if this PR will have release note line, as it changes behaviour in UX

**Comment by medinatiger** [comment]
/release-note "Allow version arg in kubeadm upgrade apply to be optional if config file already have version info"

File: cmd/kubeadm/app/cmd/upgrade/apply.go
Comment (luxas) [inline cmd/kubeadm/app/cmd/upgrade/apply.go]: Can you somehow split this out to only read the config file and determine whether it was set?
This is not correct I think, as it shouldn't ask the cluster for that information.
What we want to do here is read from the config only; and decide based on that.

**Review by luxas — CHANGES_REQUESTED** [review]
Please fix

**Comment by luxas** [comment]
ping @medinatiger

**Comment by jamiehannaford** [comment]
Hey @medinatiger, do you have any questions about this PR? We really appreciate the effort and would love to get this in.

**Comment by medinatiger** [comment]
@jamiehannaford, thanks for checking. Just back from vacation. I will find some bandwidth this week to address lucas's comment, and get this PR checked in.

File: cmd/kubeadm/app/cmd/upgrade/apply.go
Comment (medinatiger) [reply cmd/kubeadm/app/cmd/upgrade/apply.go]: @luxas I rearranged a bit, PTAL

**Comment by medinatiger** [comment]
@luxas could you please take a look for the modified PR?

File: cmd/kubeadm/app/phases/upgrade/configuration.go
Comment (luxas) [inline cmd/kubeadm/app/phases/upgrade/configuration.go]: nit: I don't think this func should log anything...

File: cmd/kubeadm/app/phases/upgrade/configuration.go
Comment (luxas) [inline cmd/kubeadm/app/phases/upgrade/configuration.go]: Is this logic necessary or is this case redundant as it's handled inside of ReadFile()?

**Review by luxas — APPROVED** [review]
Two nits only, but now this generally LGTM, thanks @medinatiger!

/approve

I'll let Tim merge this when the nits are fixed

**Comment by medinatiger** [comment]
@timothysc I fixed the nit as pointed out by Lucas (I rebase them into a single CL, sorry for the inconvenience). Please merge when you have a chance

**Review by luxas — APPROVED** [review]
/lgtm

**Pull request merged**
